src/Draco.Compiler.Tests/Syntax/LexerTests.cs

```diff
@@ -1233,19 +1233,45 @@ public void TestOperator(string text, TokenType tokenType)
     }
 
     [Theory]
-    [InlineData("0", TokenType.LiteralInteger)]
-    [InlineData("123", TokenType.LiteralInteger)]
-    [InlineData("12.3", TokenType.LiteralFloat)]
-    [InlineData("true", TokenType.KeywordTrue)]
+    [InlineData("0", 0, TokenType.LiteralInteger)]
+    [InlineData("123", 123, TokenType.LiteralInteger)]
+    [InlineData("12.3", 12.3, TokenType.LiteralFloat)]
+    [InlineData("true", null, TokenType.KeywordTrue)]
+    [InlineData("0x4c6", 1222, TokenType.LiteralInteger)]
+    [InlineData("0b110101", 53, TokenType.LiteralInteger)]
+    [InlineData("10E3", 10000d, TokenType.LiteralFloat)]
+    [InlineData("10E-3", 0.01, TokenType.LiteralFloat)]
+    [InlineData("0.1e+4", 1000d, TokenType.LiteralFloat)]
+    [InlineData("123.345E-12", 1.23345E-10, TokenType.LiteralFloat)]
     [Trait("Feature", "Literals")]
-    public void TestLiteral(string text, TokenType tokenType)
+    public void TestLiteral(string text, object? value, TokenType tokenType)
     {
         var tokens = Lex(text);
 
         AssertNextToken(tokens, out var token);
         Assert.Equal(tokenType, token.Type);
-        Assert.Equal(text, token.Text);
+        if (value is not null) Assert.Equal(value, token.Value);
+        AssertNoTriviaOrDiagnostics(token);
+
+        AssertNextToken(tokens, out token);
+        Assert.Equal(TokenType.EndOfInput, token.Type);
+        Assert.Equal(string.Empty, token.Text);
         AssertNoTriviaOrDiagnostics(token);
+    }
+
+    [Theory]
+    [InlineData("0.1e+", TokenType.LiteralFloat)]
+    [InlineData("345E-", TokenType.LiteralFloat)]
+    [Trait("Feature", "Literals")]
+    public void TestNumericLiteralInvalidNonDecimalFormats(string text, TokenType tokenType)
+    {
+        var tokens = Lex(text);
+
+        AssertNextToken(tokens, out var token);
+        Assert.Equal(tokenType, token.Type);
+        Assert.Null(token.Value);
+        Assert.Single(token.Diagnostics);
+        Assert.Equal("unexpected floating-point literal end", token.Diagnostics[0].Title);
```

Comment:
Ok, keep in mind that after the syntax tree rework the assertion will reference the diag templates

Reply:
For now this is fine